content/src/content/jcr_root/apps/core/wcm/components/modal/v1/modal/modal.html

```diff
@@ -0,0 +1,28 @@
+<!--/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  ~ Copyright 2019 Adobe Systems Incorporated
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/-->
+<div data-sly-use.modal="com.adobe.cq.wcm.core.components.models.Modal" data-sly-use.template="core/wcm/components/commons/v1/templates.html">
+    <div data-sly-test="${wcmmode.edit}">
+        <h4>Modal Id: ${modal.modalId}</h4>

```

Comment:
I would instead use the title, with:  `${modal.title || 'Dialog' @ i18n} - ${modal.id}`

Reply:
Please use `i18n` for all admin/UI text.

Reply:
I would rather we:

• not define the html tag externally [0] and add css class `cmp-modal`, the ID and all other API / data attributes on the same wrapper div as the Sling Model inclusion (as we do for all other components).
• use a placeholder to define the edit mode UI as we do with the form hidden field [1], rather than a div with h4 tags, which isn't that neat if there are quite a few of these on the page.
• on the placeholder display the dialog title if it's available, as well as the id (not the fragment path, as it isn't readable at a glance). Basically similar to @vladbailescu's suggestion above, but the 'Dialog' string comes for free in the placeholder:
<img width="817" alt="Screenshot 2019-04-23 at 19 05 42" src="https://user-images.githubusercontent.com/25616660/56604866-d584f280-65fa-11e9-835d-c20b4a41db72.png">

oversimplified example of how I would expect the HTML to look below. The `appendText` in the placeholder would define the info to display in edit mode.
```html
<div data-sly-use.modal="com.adobe.cq.wcm.core.components.models.Modal"
     data-sly-use.templates="core/wcm/components/commons/v1/templates.html"
     data-sly-test="${!wcmmode.edit}"
     id="${modal.Id}"
     class="cmp-modal"
     role="dialog"
     data-*></div>
<sly data-sly-call="${templates.placeholder @ isEmpty = wcmmode.edit, classAppend, emptyTextAppend = appendText}"></sly>

```

[0] - content/src/content/jcr_root/apps/core/wcm/components/modal/v1/modal/_cq_htmlTag/.content.xml
[1] - https://github.com/adobe/aem-core-wcm-components/blob/master/content/src/content/jcr_root/apps/core/wcm/components/form/hidden/v2/hidden/hidden.html